src/App.tsx

```diff
@@ -1,7 +1,37 @@
-import React from 'react';
+import { Component } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleDalateClick);
+  }
+
+  componentWillUnmount(): void {
```

Comment:
It's extra, bc TypeScript already knows about React types
Also,  choose just 1 way & use it, bc in 12 lines you don't do this
```suggestion
  componentWillUnmount() {
```